src/server/pmix_server.c

```diff
@@ -2200,6 +2200,7 @@ static void _mdxcbfunc(int sd, short argc, void *cbdata)
     char byte;
     bool found;
     pmix_collect_t ctype;
+    bool holding_ns_locks = false;
```

Comment:
Let's remove `holding_ns_locks`

Reply:
The reason I didnt remove `holding_ns_locks` here is for the error/early exit case before we obtain the lock. If some code path goes to `finish_collective` before acquiring the lock, then we will be unlocking something that wasn't already locked, which will in itself trigger an error.